### PR TITLE
Bugfix: nb_inventory.py group_by: tags (NB 2.9+)

### LINKED ISSUE #287: NetBox 2.9 Tags Move from List to List of Dictionaries
##### ISSUE TYPE

 - Bug Report

###### Ansible:

2.9.11

###### Netbox:

2.9 beta 1

###### Collection:

0.2.3

##### SUMMARY

Tags in NetBox 2.9 are moving data types

##### STEPS TO REPRODUCE

Create a NetBox 2.9 beta, and assign tags, tags now have a different output from NetBox

### NetBox 2.8 Response

```
      "tags": [
        "test_tag01"
      ],
```

### NetBox 2.9 Response

```
      "tags": [
        {
          "id": 1,
          "url": "https://netbox-beta/api/extras/tags/1/",
          "name": "test_tag01",
          "slug": "test_tag01",
          "color": "607d8b"
        }
      ],
```

### PR DESCRIPTION
**Pull request by jkuzma07**
Fixes #287. Code stolen from @dotSource-GmbH.
Made a mess of some authorship revisions on the #333 fork. Starting fresh.

### CODE REVIEW

**Review by FragmentedPacket — CHANGES_REQUESTED** [review]

File: plugins/inventory/nb_inventory.py [hunk 1]
```diff
@@ -586,7 +586,18 @@ def extract_primary_ip6(self, host):
...
+            # Check the type of the first element in the "tags" array.
+            # If a dictionary (Netbox >= 2.9), return an array of tags' slugs.
+            if isinstance(tag_zero, dict):
+                return list(sub["slug"] for sub in host["tags"])
+            # If a string (Netbox <= 2.8), return the original "tags" array.
+            elif isinstance(tag_zero, str):
+                return host["tags"]
+        # If tag_zero not defined (no tags), return an empty array.
+        except Exception:
+            return []
```
[the remark this answers is not in the document]
Comment (jkuzma07) [reply plugins/inventory/nb_inventory.py]: Saves memory and returns the same data? I like it.

**Comment by FragmentedPacket** [comment]
It's probably just all the stuff from Ansible 2.10 and NB2.9 changes.

I'm trying to true up all the testing before cutting a new release for things that made it through without testing

**Comment by MaartenMol** [comment]
@FragmentedPacket if I make the local changes like above code, do I have a working workaround?

**Comment by FragmentedPacket** [comment]
@MaartenMol Yes it should. I believe I cut 1.1.0 that has this fix in it as well.

You can install from galaxy or build and install locally

**Pull request merged**